### PR TITLE
CRM-16820. Check curl info and regex result.

### PR DESCRIPTION
**Pull request by xurizaemon**
- [CRM-16820: Improve error handling in DPS IPN](https://issues.civicrm.org/jira/browse/CRM-16820)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
This seems fine - but note that it's against master & this payment processor may come out of core before 4.7 is release

File: CRM/Core/Payment/PaymentExpressIPN.php [hunk 1]
```diff
@@ -297,23 +297,33 @@ public static function main($dps_method, $rawPostData, $dps_url, $dps_user, $dps
...
-        $responseText = CRM_Core_Payment_PaymentExpressUtils::_xmlElement($response, 'ResponseText');
-        $authCode = CRM_Core_Payment_PaymentExpressUtils::_xmlElement($response, 'AuthCode');
-        $DPStxnRef = CRM_Core_Payment_PaymentExpressUtils::_xmlElement($response, 'DpsTxnRef');
-        $qfKey = CRM_Core_Payment_PaymentExpressUtils::_xmlElement($response, "TxnData1");
-        $privateData = CRM_Core_Payment_PaymentExpressUtils::_xmlElement($response, "TxnData2");
-        list($component, $paymentProcessorID,) = explode(',', CRM_Core_Payment_PaymentExpressUtils::_xmlElement($response, "TxnData3"));
-        $amount = CRM_Core_Payment_PaymentExpressUtils::_xmlElement($response, "AmountSettlement");
-        $merchantReference = CRM_Core_Payment_PaymentExpressUtils::_xmlElement($response, "MerchantReference");
+        $info = curl_getinfo($curl);
+        error_log(print_r($info,1));
```
Comment (totten) [inline CRM/Core/Payment/PaymentExpressIPN.php]: @xurizaemon , agree it's good to check for success and log an error on failure.

The line above (`error_log(print_r($info,1));`) reads a bit like overkill/debugging -- wouldn't it write out the details to the error log all the time -- even when the request is successful?

Comment (xurizaemon) [inline CRM/Core/Payment/PaymentExpressIPN.php]: Yup that's the debug I was referring to in [comment on PR6415](https://github.com/civicrm/civicrm-core/pull/6415#commitcomment-12531601). This commit shouldn't have gone into that branch at all, AND it shouldn't include the debug in this one. I'm off deck a few days but will put this on radar for return.

Comment (colemanw) [inline CRM/Core/Payment/PaymentExpressIPN.php]: @xurizaemon can you fix this commit so we can merge this in?

Comment (eileenmcnaughton) [inline CRM/Core/Payment/PaymentExpressIPN.php]: I think we will probably remove that processor from 4.7 actually

**Comment by davecivicrm** [comment]
Closing based on @eileenmcnaughton  comment above.

**Comment by eileenmcnaughton** [comment]
@davecivicrm at this rate it might be 4.8 rather than 4.7 by the time we remove it & go extension - but I think this is OK to close unless Chris says otherwise

**Comment by davecivicrm** [comment]
@eileenmcnaughton Ok, have him reopen if needed. Separately, can you pls get back to me on whether the new unwanted behavior in back-office contribution form to display "(Contribution Amount - 1)" in selector is related to your refactoring work or not. If not, I'll get someone on our team to investigate.
screen shot 2015-09-30 at 10 53 19 pm

**Comment by eileenmcnaughton** [comment]
It probably is - I'm just trying to find time to look at it.

**Pull request closed without merging**